### PR TITLE
Product Sequence

### PR DESCRIPTION
**Pull request by gdgellatly**
Add product template default code related field as required
to stop unfriendly errors in UI
Add tests

### CODE REVIEW

File: product_sequence/tests/test_product_sequence.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#
+##############################################################################
+
+from openerp.tests.common import TransactionCase
+from psycopg2 import IntegrityError
+
+
+class ProductSequenceCase(TransactionCase):
+
+    def setUp(self):
```
Comment (lmignon) [inline product_sequence/tests/test_product_sequence.py:29]: no really required

**Comment by lmignon** [comment]
:+1:  Thank you for the tests!

File: product_sequence/tests/test_product_sequence.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        super(ProductSequenceCase, self).setUp()
+
+    def test_copy(self):
+        original = self.env.ref('product.product_product_1')
+        new = original.copy()
+        self.assertFalse(original.default_code == new.default_code)
```
Comment (Cedric-Pigeon) [inline product_sequence/tests/test_product_sequence.py:35]: Why don't you use <code>self.assertNotEqual</code> ?

File: product_sequence/tests/test_product_sequence.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        self.assertTrue(new.default_code)
+
+    def test_write_slash(self):
+        product = self.env.ref('product.product_product_1')
+        self.cr.execute("""
+            UPDATE product_product
+              SET default_code = '/'
+              WHERE id=%s""", (product.id,))
+        product.write({'name': 'Test'})
+        self.assertFalse(product.default_code == '/')
```
Comment (Cedric-Pigeon) [inline product_sequence/tests/test_product_sequence.py:49]: Same remark.

**Comment by Cedric-Pigeon** [comment]
Some small remarks, thanks for the test.

**Comment by gdgellatly** [comment]
@pedrobaeza same with this one.  I never used this module, all I was doing was making tests works and greenifying it.  Happy to merge, happy to close.

**Review by pedrobaeza — APPROVED** [review]
Well, as this is only for greenifying branch, I'm not going to be picky this time, although the comments are valid improvements. Fast-tracking for solving this.

**Pull request merged**